generator/sbpg/targets/rust.py

```diff
@@ -101,11 +122,37 @@ def parse_type(field):
     # This is an inner class, call default constructor
     return "%s::parse(_buf)" % field.type_id
 
+def gps_time(msg, all_messages):
+  def time_aware_header(type_id):
+    for m in all_messages:
+      if m.identifier == type_id:
+        return any([f.identifier == "t" for f in m.fields])
+    return False
+
+  has_tow = False
+  has_wn = False
+
+  for f in msg.fields:
+    if f.identifier == "header" and time_aware_header(f.type_id):
+      return GPS_TIME_HEADER
+    elif f.identifier == "tow":
+      has_tow = True
+    elif f.identifier == "wn":
+      has_wn = True
+
+  if has_tow and has_wn:
+    return GPS_TIME
+  elif has_tow:
+    return GPS_TIME_ONLY_TOW
+  else:
+    return 'None'
```

Comment:
Not sure if this is the best way to determine which messages are time-aware/how that time is stored but it seems to work so far

Reply:
I think this is a good starting point at the very least. I'd have to peruse all the message definitions to figure out if there's other ways we include a GPS time in messages, but this should cover the vast majority if not all of them.

I think it should be noted that the time values can have different meanings depending on the message type they're in. For instance the time value in `MSG_OBS` is the time that the observations were made, but in `MSG_EPHEMERIS_GPS` (or any ephemeris message) it's a center point in the period of time the data is valid so it is typically several minutes to a couple of hours in the future from when the message was actually generated. We can discuss offline if this is a problem for what you're working on.

Reply:
For this use case the "time of event" or "time of occurrence" is the meaning of `.gps_time()` -- that is, we want to define a linear time ordering of the messages for use in things like the "sbp chopper" and the interleaver.

I talked with @notoriaga about this briefly and I agree that "t"/"wn"/"tow" covers most of the use case correctly, but we should do an audit of everything that gets grabbed by this and then maybe introduce a "time_ordering" property that we can expose to the yaml spec for this purpose.